### PR TITLE
[interpreter] Fix LLVM target selection

### LINKED ISSUE #14075: [FreeBSD] root.exe crash at end of compilation during modules.idx generation
### Check duplicate issues.

- [X] Checked for duplicates

### Description

as in the description. the root process crashes somewhere in ctor of Interpreter.
Below the stack trace (plus some printouts of the interpreter args as I have no debug info in cling atm, build takes too many resources):
````
[mkrzewic@hack ~/build/root]$ lldb bin/root.exe
(lldb) target create "bin/root.exe"
Current executable set to '/home/mkrzewic/build/root/bin/root.exe' (x86_64).
(lldb) run
Process 35096 launched: '/home/mkrzewic/build/root/bin/root.exe' (x86_64)
interpArgs.size(): 20
interpArgs[0]: cling4root
interpArgs[1]: -DNDEBUG
interpArgs[2]: -UDEBUG
interpArgs[3]: -U_DEBUG
interpArgs[4]: -I/usr/home/mkrzewic/build/root/etc/
interpArgs[5]: -I/usr/home/mkrzewic/build/root/etc//cling
interpArgs[6]: -I/usr/home/mkrzewic/build/root/etc//cling/plugins/include
interpArgs[7]: -I/usr/home/mkrzewic/build/root/include/
interpArgs[8]: -Wno-undefined-inline
interpArgs[9]: -fsigned-char
interpArgs[10]: -O1
interpArgs[11]: -mllvm
interpArgs[12]: -optimize-regalloc=0
interpArgs[13]: -fmodule-map-file=/usr/home/mkrzewic/build/root/include//module.modulemap
interpArgs[14]: -fmodules-cache-path=/usr/home/mkrzewic/build/root/lib
interpArgs[15]: -fmodules
interpArgs[16]: -fno-implicit-module-maps
interpArgs[17]: -Rmodule-build
interpArgs[18]: -fno-autolink
interpArgs[19]: --ptrcheck
Process 35096 stopped
* thread #1, name = 'root.exe', stop reason = signal SIGABRT
frame #0: 0x00000008279b9dca libc.so.7`__sys_thr_kill + 10
libc.so.7`__sys_thr_kill:
->  0x8279b9dca <+10>: jb     0x8279b7f28
0x8279b9dd0 <+16>: retq
0x8279b9dd1:       int3
0x8279b9dd2:       int3
(lldb) bt
* thread #1, name = 'root.exe', stop reason = signal SIGABRT
* frame #0: 0x00000008279b9dca libc.so.7`__sys_thr_kill + 10
frame #1: 0x00000008279326f4 libc.so.7`__raise + 52
frame #2: 0x00000008279e3bc9 libc.so.7`abort + 73
frame #3: 0x00000008248c0e39 libcxxrt.so.1`___lldb_unnamed_symbol472 + 73
frame #4: 0x0000000825448abf libc++.so.1`operator new(unsigned long) + 111
frame #5: 0x000000082544f2e3 libc++.so.1`std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::operator=(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) + 195
frame #6: 0x00000008351c7d9c libCling.so`llvm::DataLayout::operator=(llvm::DataLayout const&) + 44
frame #7: 0x00000008351c7cd3 libCling.so`llvm::DataLayout::DataLayout(llvm::DataLayout const&) + 195
frame #8: 0x00000008351cbc62 libCling.so`cling::IncrementalJIT::IncrementalJIT(cling::IncrementalExecutor&, clang::CompilerInstance const&, std::__1::unique_ptr<llvm::orc::ExecutorProcessControl, std::__1::default_delete<llvm::orc::ExecutorProcessControl> >, llvm::Error&, void*, bool) + 1154
frame #9: 0x00000008351c863a libCling.so`cling::IncrementalExecutor::IncrementalExecutor(clang::DiagnosticsEngine&, clang::CompilerInstance const&, void*, bool) +378
frame #10: 0x0000000835161b6b libCling.so`cling::Interpreter::Interpreter(int, char const* const*, char const*, std::__1::vector<std::__1::shared_ptr<clang::ModuleFileExtension>, std::__1::allocator<std::__1::shared_ptr<clang::ModuleFileExtension> > > const&, void*, bool, cling::Interpreter const*) + 2107
frame #11: 0x000000083506c054 libCling.so`TCling::TCling(char const*, char const*, char const* const*, void*) [inlined] cling::Interpreter::Interpreter(this=0x000000082eccac00, argc=20, argv=0x000000082edb3000, llvmdir=<unavailable>, moduleExtensions=0x0000000820b5e5d0, extraLibHandle=0x000000082ddf0808, noRuntime=false) at Interpreter.h:367:11
frame #12: 0x000000083506c03c libCling.so`TCling::TCling(char const*, char const*, char const* const*, void*) [inlined] std::__1::__unique_if<cling::Interpreter>::__unique_single std::__1::make_unique<cling::Interpreter, unsigned long, char const**, TString&, std::__1::vector<std::__1::shared_ptr<clang::ModuleFileExtension>, std::__1::allocator<std::__1::shared_ptr<clang::ModuleFileExtension> > >&, void*&>(__args=<unavailable>, __args=<unavailable>, __args=0x0000000820b5e568, __args=0x0000000820b5e5d0, __args=<unavailable>) at unique_ptr.h:725:32
frame #13: 0x000000083506c00f libCling.so`TCling::TCling(this=0x000000082ecc4600, name=<unavailable>, title=<unavailable>, argv=<unavailable>, interpLibHandle=0x000000082ddf0808) at TCling.cxx:1546:19
frame #14: 0x0000000835066ec9 libCling.so`::CreateInterpreter(interpLibHandle=0x000000082ddf0808, argv=0x0000000820b5e8a0) at TCling.cxx:609:22
frame #15: 0x0000000822e85418 libCore.so`TROOT::InitInterpreter(this=0x0000000823028a50) at TROOT.cxx:2087:19
frame #16: 0x0000000822e85333 libCore.so`ROOT::Internal::GetROOT2() at TROOT.cxx:389:22
frame #17: 0x0000000822e3a23a libCore.so`TApplication::TApplication(this=0x000000082ece43c0, appClassName="Rint", argc=0x0000000820b60a9c, argv=0x0000000820b60b18, (null)=<unavailable>, numOptions=0) at TApplication.cxx:153:9
frame #18: 0x0000000821cbcccb libRint.so`TRint::TRint(this=0x000000082ece43c0, appClassName="Rint", argc=0x0000000820b60a9c, argv=0x0000000820b60b18, options=0x0000000000000000, numOptions=0, noLogo=<unavailable>, exitOnUnknownArgs=<unavailable>) at TRint.cxx:173:6
frame #19: 0x00000000002020d1 root.exe`main(argc=1, argv=0x0000000820b60b18) at rmain.cxx:80:24
frame #20: 0x0000000000201d60 root.exe`_start(ap=<unavailable>, cleanup=<unavailable>) at crt1_c.c:75:7
(lldb)
````

### Reproducer

cmake ../../src/root -DCMAKE_CXX_STANDARD=17 -DCMAKE_BUILD_TYPE=RelWithDebInfo

### ROOT version

recent master (@bce645)

### Installation method

source

### Operating system

FreeBSD 13.2-RELEASE-p5

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hahnjo**
@mkrzewic @eamjensen could you maybe give this a short test on FreeBSD? I hope that a "vanilla" configuration of ROOT should now succeed, without specifying any `LLVM_TARGETS_TO_BUILD`.

Closes #14075

### CODE REVIEW

**Comment by mkrzewic** [comment]
@hahnjo configure seems to finish without errors, but it still seems to target all possible archs:

```
-- Targeting AArch64
-- Targeting AMDGPU
-- Targeting ARM
-- Targeting AVR
-- Targeting BPF
-- Targeting Hexagon
-- Targeting Lanai
-- Targeting Mips
-- Targeting MSP430
-- Targeting NVPTX
-- Targeting PowerPC
-- Targeting RISCV
-- Targeting Sparc
-- Targeting SystemZ
-- Targeting WebAssembly
-- Targeting X86
-- Targeting XCore
-- Clang version: 13.0.0
-- Not building amdgpu-arch: hsa-runtime64 not found
-- Building Cling as part of ROOT
-- Cling version (from VERSION file): ROOT_1.0~dev
```

build ongoing

**Comment by hahnjo** [comment]
> it still seems to target all possible archs

Uh, I missed the most important case of `freebsdamd64`. Should be fixed with the most recent push, maybe you can try another round of (only) `cmake` after the build finished.

**Comment by mkrzewic** [comment]
@hahnjo configure seems to work now:

```
-- LLVM host triple: x86_64-unknown-freebsd13.2
-- LLVM default target triple: x86_64-unknown-freebsd13.2
-- Building with -fPIC
-- Found Python3: /usr/local/bin/python3 (found suitable version "3.9.18", minimum required is "3.0") found components: Interpreter 
-- Targeting X86
-- Targeting NVPTX
-- Clang version: 13.0.0
```
the build will take a while.
thanks

**Comment by mkrzewic** [comment]
all good! builds and works.

**Comment by hahnjo** [comment]
@mkrzewic thanks for testing!

File: interpreter/CMakeLists.txt [hunk 1]
```diff
@@ -81,21 +81,17 @@ set(LLVM_ABI_BREAKING_CHECKS FORCE_OFF CACHE BOOL "" FORCE)
...
   set(ROOT_CLING_TARGET "SystemZ")
-elseif(ROOT_ARCHITECTURE MATCHES linuxriscv64)
+elseif(ROOT_ARCHITECTURE MATCHES riscv64)
   set(ROOT_CLING_TARGET "RISCV")
-elseif(ROOT_ARCHITECTURE MATCHES linux)
-  set(ROOT_CLING_TARGET "X86")
-elseif(ROOT_ARCHITECTURE MATCHES macosxarm64)
-  set(ROOT_CLING_TARGET "AArch64")
-elseif(ROOT_ARCHITECTURE MATCHES macosx)
+elseif(ROOT_ARCHITECTURE MATCHES freebsd OR ROOT_ARCHITECTURE MATCHES linux OR ROOT_ARCHITECTURE MATCHES macosx)
```
Comment (pcanal) [inline interpreter/CMakeLists.txt:94]: Why is `macosxarm64` no longer needed?

Comment (hahnjo) [reply interpreter/CMakeLists.txt:94]: It's taken care of by `ROOT_ARCHITECTURE MATCHES arm64`, which relieves us from the pain to specify every single platform cross architecture combination.

Comment (pcanal) [reply interpreter/CMakeLists.txt:94]: That would be good but when does it change to `arm64`?  I see https://github.com/hahnjo/root/blob/interpreter-targets/cmake/modules/SetUpMacOS.cmake which sets it to the longer name.

**Review by pcanal — APPROVED** [review]
LGTM.

**Pull request merged**